src/lazy/binary/raw/v1_1/struct.rs

```diff
@@ -117,20 +120,167 @@ impl<'top> LazyRawStruct<'top, BinaryEncoding_1_1> for LazyRawBinaryStruct_1_1<'
     }
 }
 
+enum StructType {
+    Invalid,
+    FlexSym,
+    SymbolAddress,
+}
+
 pub struct RawBinaryStructIterator_1_1<'top> {
     source: ImmutableBuffer<'top>,
+    bytes_to_skip: usize,
+    struct_type: StructType,
 }
 
 impl<'top> RawBinaryStructIterator_1_1<'top> {
-    pub(crate) fn new(input: ImmutableBuffer<'top>) -> RawBinaryStructIterator_1_1<'top> {
-        RawBinaryStructIterator_1_1 { source: input }
+    pub(crate) fn new(
+        opcode_type: OpcodeType,
+        input: ImmutableBuffer<'top>,
+    ) -> RawBinaryStructIterator_1_1<'top> {
+        RawBinaryStructIterator_1_1 {
+            source: input,
+            bytes_to_skip: 0,
+            struct_type: match opcode_type {
+                OpcodeType::StructSymAddress => StructType::SymbolAddress,
+                OpcodeType::StructFlexSym => StructType::FlexSym,
+                _ => StructType::Invalid,
+            },
+        }
+    }
+
+    /// Helper function called by [`Self::peek_field`] in order to parse a FlexSym encoded
+    /// struct field names. If no field is available, None is returned, otherwise the symbol and an
+    /// [`ImmutableBuffer`] positioned after the field name is returned.
+    ///
+    /// The opcode variant of the FlexSym is not currently implemented.
+    fn peek_field_flexsym(
+        buffer: ImmutableBuffer<'top>,
+    ) -> IonResult<Option<(LazyRawBinaryFieldName_1_1<'top>, ImmutableBuffer<'top>)>> {
+        use crate::IonError;
+        use std::cmp::Ordering;
+
+        if buffer.is_empty() {
+            return Ok(None);
+        }
+
+        let (flex_sym, after) = buffer.read_flex_int()?;
+        let sym_value = flex_sym.value();
+        let (sym, after) = match sym_value.cmp(&0) {
```

Comment:
Could you move this bit of logic to [the `FlexSym` type](https://github.com/amazon-ion/ion-rust/blob/9fd213c591c8cf95a5cf50b6f0f2f9fc5908a0b4/src/lazy/encoder/binary/v1_1/flex_sym.rs#L10)?